src/stable_hasher.rs

```diff
@@ -0,0 +1,99 @@
+use std::hash::Hasher;
+
+/// Wrapper over a hasher that provides stable output across platforms
+/// Based on https://github.com/rust-lang/rust/blob/c0955a34bcb17f0b31d7b86522a520ebe7fa93ac/src/librustc_data_structures/stable_hasher.rs#L78-L166
+///
+/// To that end we always convert integers to little-endian format before
+/// hashing and the architecture dependent `isize` and `usize` types are
+/// extended to 64 bits if needed.
+pub struct StableHasher {
+    /// Using xxh3-64 with default seed/secret as the portable hasher.
+    state: xxhash_rust::xxh3::Xxh3,
+}
+
+impl StableHasher {
+    #[inline]
+    pub fn new() -> Self {
+        Self {
+            state: xxhash_rust::xxh3::Xxh3::new(),
+        }
+    }
+}
+
+impl Hasher for StableHasher {
```

Comment:
This is fantastic stuff. How are you feeling about this PR?

Reply:
I think it might be good to go now :+1:

Reply:
I may have one more change. I'll push shortly.

Reply:
I just reviewed the changes and things are looking great. You comfortable with me merging?

Reply:
:shipit: !

Reply:
I also just tested the benchmarks and the numbers are looking way better. Suprisingly consistent across different failure rates, which is a great sign.

Reply:
Awesome. Could you make a crates.io release with this new version?

Reply:
For sure. Just got off work so I'll merge and prep for all of that